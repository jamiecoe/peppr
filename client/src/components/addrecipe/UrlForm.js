import React, { Component } from 'react';
import { Field, reduxForm } from 'redux-form';
import { connect } from 'react-redux';
import { checkUrl } from '../../actions/recipes';

import goodfood from '../../assets/images/goodfood.png';
import delicious from '../../assets/images/delicious.png';
import jamieoliver from '../../assets/images/jamieoliver.png';

export class UrlForm extends Component {
  onSubmit(values) {
    this.props.checkUrl(values);
  }

  renderAlert() {
    if (this.props.error) {
      return (
        <p className="urlform__input--errortext">
          <strong>Oops!</strong> {this.props.error}
        </p>
      );
    }
    return null;
  }

  renderField(field) {
    const { touched, error } = field.meta;

    return (
      <div>
        <div className="urlform addrecipe__innermargin">
          {/* <label>{field.label}</label> */}
          <input
            className="urlform__input"
            type="url"
            {...field.input}
            placeholder="Enter URL"
            aria-label={field.arialabel}
          />
          <button type="submit" className="urlform__btn">
            <i className="ion-ios-download urlform__btn__icon" />
          </button>
        </div>
        <p className="urlform__input--errortext addrecipe__innermargin">
          {touched ? error : ''}
        </p>
      </div>
    );
  }

  render() {
    const { handleSubmit } = this.props;
    return (
      <form onSubmit={handleSubmit(this.onSubmit.bind(this))}>
        <p className="urlform__text addrecipe__innermargin">
          Enter a recipe URL from one of our partner sites to save the recipe
          automatically:
        </p>
        <Field name="url" arialabel="Url" component={this.renderField} />
        {this.renderAlert()}
        <div className="urlform__modal">
          <p className="addrecipe__innermargin addrecipe__partnersite--text">
            Our partner sites are:
          </p>
          <div className="urlform__modal__imagecontainer addrecipe__innermargin">
            <a href="https://www.bbcgoodfood.com/" target="_blank" rel="noopener noreferrer">
              <img src={goodfood} alt="BBC Good Food logo" />
            </a>
            <a href="https://www.jamieoliver.com/" target="_blank" rel="noopener noreferrer">
              <img src={jamieoliver} alt="Jamie Oliver logo" />
            </a>
            <a href="http://www.deliciousmagazine.co.uk/" target="_blank" rel="noopener noreferrer">
              <img src={delicious} alt="Delicious Magazine logo" />
            </a>
          </div>
        </div>
      </form>
    );
  }
}

const validate = (values) => {
  const errors = {};

  if (!values.url) {
    errors.url = 'Enter a recipe URL';
  }

  return errors;
};

const mapStateToProps = ({ newRecipe, error }) => ({
  newRecipe,
  error,
});

export default reduxForm({
  validate,
  form: 'UrlForm',
})(connect(mapStateToProps, { checkUrl })(UrlForm));
