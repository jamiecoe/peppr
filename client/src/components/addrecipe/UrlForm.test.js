import React from 'react';
import { shallow } from 'enzyme';
import { UrlForm, validate } from './UrlForm';

describe('UrlForm', () => {
  const mockCheckUrl = jest.fn();
  const mockHandleSubmit = jest.fn(fn => fn);
  const props = {
    error: '',
    checkUrl: mockCheckUrl,
    handleSubmit: mockHandleSubmit,
  };
  let urlForm = shallow(<UrlForm {...props} />);

  it('should render properly', () => {
    expect(urlForm).toMatchSnapshot();
  });

  describe('when user submits form', () => {
    beforeEach(() => {
      urlForm.find('form').simulate('submit');
    });

    it('should call mockHandleSubmit()', () => {
      expect(mockHandleSubmit).toHaveBeenCalled();
    });

    it('should call mockCheckUrl()', () => {
      expect(mockCheckUrl).toHaveBeenCalled();
    });
  });

  describe('when there is an error on props', () => {
    beforeEach(() => {
      props.error = 'this is a test error';
      urlForm = shallow(<UrlForm {...props} />);
    });

    it('should render the error message from props', () => {
      expect(urlForm.find('.urlform__input--errortext').text()).toBe(
        `Oops! ${props.error}`,
      );
    });
  });

  describe('when the `renderField` method is invoked', () => {
    const fieldProps = {
      meta: {
        touched: true,
        error: 'test error',
      },
      textfield: true,
      arialabel: 'test arialabel',
      input: {},
    };

    const RenderedField = urlForm.instance().renderField(fieldProps);
    const renderField = shallow(RenderedField);

    it('should render properly', () => {
      expect(renderField).toMatchSnapshot();
    });

    it('should include an error message', () => {
      expect(
        renderField
          .find('.urlform__input--errortext.addrecipe__innermargin')
          .text(),
      ).toBe(fieldProps.meta.error);
    });
  });
});

describe('validate function', () => {
  it('should return the correct errors, if the values are missing', () => {
    const mockValues = {};

    const expectedError = {
      url: 'Enter a recipe URL',
    };

    expect(validate(mockValues)).toEqual(expectedError);
  });
});
