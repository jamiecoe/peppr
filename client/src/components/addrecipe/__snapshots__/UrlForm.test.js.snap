// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlForm should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UrlForm
    checkUrl={[Function]}
    error=""
    handleSubmit={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="urlform__text addrecipe__innermargin"
>
          Enter a recipe URL from one of our partner sites to save the recipe automatically:
</p>,
        <Field
          arialabel="Url"
          component={[Function]}
          name="url"
/>,
        null,
        <div
          className="urlform__modal"
>
          <p
                    className="addrecipe__innermargin addrecipe__partnersite--text"
          >
                    Our partner sites are:
          </p>
          <div
                    className="urlform__modal__imagecontainer addrecipe__innermargin"
          >
                    <a
                              href="https://www.bbcgoodfood.com/"
                              rel="noopener noreferrer"
                              target="_blank"
                    >
                              <img
                                        alt="BBC Good Food logo"
                                        src="goodfood.png"
                              />
                    </a>
                    <a
                              href="https://www.jamieoliver.com/"
                              rel="noopener noreferrer"
                              target="_blank"
                    >
                              <img
                                        alt="Jamie Oliver logo"
                                        src="jamieoliver.png"
                              />
                    </a>
                    <a
                              href="http://www.deliciousmagazine.co.uk/"
                              rel="noopener noreferrer"
                              target="_blank"
                    >
                              <img
                                        alt="Delicious Magazine logo"
                                        src="delicious.png"
                              />
                    </a>
          </div>
</div>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Enter a recipe URL from one of our partner sites to save the recipe automatically:",
          "className": "urlform__text addrecipe__innermargin",
        },
        "ref": null,
        "rendered": "Enter a recipe URL from one of our partner sites to save the recipe automatically:",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "arialabel": "Url",
          "component": [Function],
          "name": "url",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="addrecipe__innermargin addrecipe__partnersite--text"
>
              Our partner sites are:
</p>,
            <div
              className="urlform__modal__imagecontainer addrecipe__innermargin"
>
              <a
                            href="https://www.bbcgoodfood.com/"
                            rel="noopener noreferrer"
                            target="_blank"
              >
                            <img
                                          alt="BBC Good Food logo"
                                          src="goodfood.png"
                            />
              </a>
              <a
                            href="https://www.jamieoliver.com/"
                            rel="noopener noreferrer"
                            target="_blank"
              >
                            <img
                                          alt="Jamie Oliver logo"
                                          src="jamieoliver.png"
                            />
              </a>
              <a
                            href="http://www.deliciousmagazine.co.uk/"
                            rel="noopener noreferrer"
                            target="_blank"
              >
                            <img
                                          alt="Delicious Magazine logo"
                                          src="delicious.png"
                            />
              </a>
</div>,
          ],
          "className": "urlform__modal",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Our partner sites are:",
              "className": "addrecipe__innermargin addrecipe__partnersite--text",
            },
            "ref": null,
            "rendered": "Our partner sites are:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  href="https://www.bbcgoodfood.com/"
                  rel="noopener noreferrer"
                  target="_blank"
>
                  <img
                                    alt="BBC Good Food logo"
                                    src="goodfood.png"
                  />
</a>,
                <a
                  href="https://www.jamieoliver.com/"
                  rel="noopener noreferrer"
                  target="_blank"
>
                  <img
                                    alt="Jamie Oliver logo"
                                    src="jamieoliver.png"
                  />
</a>,
                <a
                  href="http://www.deliciousmagazine.co.uk/"
                  rel="noopener noreferrer"
                  target="_blank"
>
                  <img
                                    alt="Delicious Magazine logo"
                                    src="delicious.png"
                  />
</a>,
              ],
              "className": "urlform__modal__imagecontainer addrecipe__innermargin",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="BBC Good Food logo"
                    src="goodfood.png"
/>,
                  "href": "https://www.bbcgoodfood.com/",
                  "rel": "noopener noreferrer",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "BBC Good Food logo",
                    "src": "goodfood.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="Jamie Oliver logo"
                    src="jamieoliver.png"
/>,
                  "href": "https://www.jamieoliver.com/",
                  "rel": "noopener noreferrer",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Jamie Oliver logo",
                    "src": "jamieoliver.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="Delicious Magazine logo"
                    src="delicious.png"
/>,
                  "href": "http://www.deliciousmagazine.co.uk/",
                  "rel": "noopener noreferrer",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Delicious Magazine logo",
                    "src": "delicious.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="urlform__text addrecipe__innermargin"
>
            Enter a recipe URL from one of our partner sites to save the recipe automatically:
</p>,
          <Field
            arialabel="Url"
            component={[Function]}
            name="url"
/>,
          null,
          <div
            className="urlform__modal"
>
            <p
                        className="addrecipe__innermargin addrecipe__partnersite--text"
            >
                        Our partner sites are:
            </p>
            <div
                        className="urlform__modal__imagecontainer addrecipe__innermargin"
            >
                        <a
                                    href="https://www.bbcgoodfood.com/"
                                    rel="noopener noreferrer"
                                    target="_blank"
                        >
                                    <img
                                                alt="BBC Good Food logo"
                                                src="goodfood.png"
                                    />
                        </a>
                        <a
                                    href="https://www.jamieoliver.com/"
                                    rel="noopener noreferrer"
                                    target="_blank"
                        >
                                    <img
                                                alt="Jamie Oliver logo"
                                                src="jamieoliver.png"
                                    />
                        </a>
                        <a
                                    href="http://www.deliciousmagazine.co.uk/"
                                    rel="noopener noreferrer"
                                    target="_blank"
                        >
                                    <img
                                                alt="Delicious Magazine logo"
                                                src="delicious.png"
                                    />
                        </a>
            </div>
</div>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enter a recipe URL from one of our partner sites to save the recipe automatically:",
            "className": "urlform__text addrecipe__innermargin",
          },
          "ref": null,
          "rendered": "Enter a recipe URL from one of our partner sites to save the recipe automatically:",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "arialabel": "Url",
            "component": [Function],
            "name": "url",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="addrecipe__innermargin addrecipe__partnersite--text"
>
                Our partner sites are:
</p>,
              <div
                className="urlform__modal__imagecontainer addrecipe__innermargin"
>
                <a
                                href="https://www.bbcgoodfood.com/"
                                rel="noopener noreferrer"
                                target="_blank"
                >
                                <img
                                                alt="BBC Good Food logo"
                                                src="goodfood.png"
                                />
                </a>
                <a
                                href="https://www.jamieoliver.com/"
                                rel="noopener noreferrer"
                                target="_blank"
                >
                                <img
                                                alt="Jamie Oliver logo"
                                                src="jamieoliver.png"
                                />
                </a>
                <a
                                href="http://www.deliciousmagazine.co.uk/"
                                rel="noopener noreferrer"
                                target="_blank"
                >
                                <img
                                                alt="Delicious Magazine logo"
                                                src="delicious.png"
                                />
                </a>
</div>,
            ],
            "className": "urlform__modal",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Our partner sites are:",
                "className": "addrecipe__innermargin addrecipe__partnersite--text",
              },
              "ref": null,
              "rendered": "Our partner sites are:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    href="https://www.bbcgoodfood.com/"
                    rel="noopener noreferrer"
                    target="_blank"
>
                    <img
                                        alt="BBC Good Food logo"
                                        src="goodfood.png"
                    />
</a>,
                  <a
                    href="https://www.jamieoliver.com/"
                    rel="noopener noreferrer"
                    target="_blank"
>
                    <img
                                        alt="Jamie Oliver logo"
                                        src="jamieoliver.png"
                    />
</a>,
                  <a
                    href="http://www.deliciousmagazine.co.uk/"
                    rel="noopener noreferrer"
                    target="_blank"
>
                    <img
                                        alt="Delicious Magazine logo"
                                        src="delicious.png"
                    />
</a>,
                ],
                "className": "urlform__modal__imagecontainer addrecipe__innermargin",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt="BBC Good Food logo"
                      src="goodfood.png"
/>,
                    "href": "https://www.bbcgoodfood.com/",
                    "rel": "noopener noreferrer",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "BBC Good Food logo",
                      "src": "goodfood.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt="Jamie Oliver logo"
                      src="jamieoliver.png"
/>,
                    "href": "https://www.jamieoliver.com/",
                    "rel": "noopener noreferrer",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "Jamie Oliver logo",
                      "src": "jamieoliver.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt="Delicious Magazine logo"
                      src="delicious.png"
/>,
                    "href": "http://www.deliciousmagazine.co.uk/",
                    "rel": "noopener noreferrer",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "Delicious Magazine logo",
                      "src": "delicious.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "a",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`UrlForm when the \`renderField\` method is invoked should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div>
    <div
        className="urlform addrecipe__innermargin"
    >
        <input
            aria-label="test arialabel"
            className="urlform__input"
            placeholder="Enter URL"
            type="url"
        />
        <button
            className="urlform__btn"
            type="submit"
        >
            <i
                className="ion-ios-download urlform__btn__icon"
            />
        </button>
    </div>
    <p
        className="urlform__input--errortext addrecipe__innermargin"
    >
        test error
    </p>
</div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="urlform addrecipe__innermargin"
>
          <input
                    aria-label="test arialabel"
                    className="urlform__input"
                    placeholder="Enter URL"
                    type="url"
          />
          <button
                    className="urlform__btn"
                    type="submit"
          >
                    <i
                              className="ion-ios-download urlform__btn__icon"
                    />
          </button>
</div>,
        <p
          className="urlform__input--errortext addrecipe__innermargin"
>
          test error
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              aria-label="test arialabel"
              className="urlform__input"
              placeholder="Enter URL"
              type="url"
/>,
            <button
              className="urlform__btn"
              type="submit"
>
              <i
                            className="ion-ios-download urlform__btn__icon"
              />
</button>,
          ],
          "className": "urlform addrecipe__innermargin",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "test arialabel",
              "className": "urlform__input",
              "placeholder": "Enter URL",
              "type": "url",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="ion-ios-download urlform__btn__icon"
/>,
              "className": "urlform__btn",
              "type": "submit",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "ion-ios-download urlform__btn__icon",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test error",
          "className": "urlform__input--errortext addrecipe__innermargin",
        },
        "ref": null,
        "rendered": "test error",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="urlform addrecipe__innermargin"
>
            <input
                        aria-label="test arialabel"
                        className="urlform__input"
                        placeholder="Enter URL"
                        type="url"
            />
            <button
                        className="urlform__btn"
                        type="submit"
            >
                        <i
                                    className="ion-ios-download urlform__btn__icon"
                        />
            </button>
</div>,
          <p
            className="urlform__input--errortext addrecipe__innermargin"
>
            test error
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                aria-label="test arialabel"
                className="urlform__input"
                placeholder="Enter URL"
                type="url"
/>,
              <button
                className="urlform__btn"
                type="submit"
>
                <i
                                className="ion-ios-download urlform__btn__icon"
                />
</button>,
            ],
            "className": "urlform addrecipe__innermargin",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "test arialabel",
                "className": "urlform__input",
                "placeholder": "Enter URL",
                "type": "url",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="ion-ios-download urlform__btn__icon"
/>,
                "className": "urlform__btn",
                "type": "submit",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "ion-ios-download urlform__btn__icon",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test error",
            "className": "urlform__input--errortext addrecipe__innermargin",
          },
          "ref": null,
          "rendered": "test error",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
